google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -46,59 +50,82 @@ def format_stackdriver_json(record, message):
     return json.dumps(payload)
 
 
-def get_trace_id_from_flask():
-    """Get trace_id from flask request headers.
+def get_request_data_from_flask():
+    """Get trace_id and http_request data from flask request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        dict: data about the associated http request.
     """
     if flask is None or not flask.request:
-        return None
+        return None, None
+
+    # build http_request
+    http_request = {
```

Comment:
You might be able to get or be able to infer other fields as mentioned by the HttpRequest [spec here](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest), e.g. `protocol` 

Caveat: Im not familiar with what's exposed in request types in django/flask

Reply:
Some of them I can't include, because the request isn't finished (responseSize, status, etc). Others I'm not quite sure about, like the cache fields. Do you know if that should be in the headers somewhere?

For some reason Flask didn't make the request's protocol as easy to find as the others, but it should be right there in the header, so I'll take another look there